### PR TITLE
Bump PG versions used in CI

### PR DESCRIPTION
**Pull request by fabriziomello**
Use newly released 13.14, 14.11, 15.6 and 16.2 in CI.

Disable-check: force-changelog-file

### CODE REVIEW

File: .github/ci_settings.py [hunk 1]
```diff
@@ -14,19 +14,19 @@
 # move to a new minor version below
```
Comment (jnidzwetzki) [inline .github/ci_settings.py:14]: Could you enable the upload, as mentioned in the comment? It should work now with the new PostgreSQL version.

**Review by jnidzwetzki — APPROVED** [review]

File: .github/ci_settings.py (hunk 1 above)
Comment (fabriziomello) [reply .github/ci_settings.py]: Fixed

**Comment by svenklemm** [comment]
I think pg_dump_unprivileged is gonna need adjustments

**Comment by fabriziomello** [comment]
> I think pg_dump_unprivileged is gonna need adjustments

Thanks for fixing it! Will wait for new postgres docker images be available to merge this PR.

**Pull request merged**